core/key.go

```diff
@@ -210,6 +218,22 @@ func NewCollectionSchemaVersionKey(schemaVersionId string) CollectionSchemaVersi
 	return CollectionSchemaVersionKey{SchemaVersionId: schemaVersionId}
 }
 
+func NewCollectionIndexKey(colID, name string) CollectionIndexKey {
+	return CollectionIndexKey{CollectionID: colID, IndexName: name}
+}
+
+func NewCollectionIndexKeyFromString(key string) (CollectionIndexKey, error) {
+	keyArr := strings.Split(key, "/")
+	if len(keyArr) < 4 || len(keyArr) > 5 || keyArr[1] != "collection" || keyArr[2] != "index" {
+		return CollectionIndexKey{}, errors.WithStack(ErrInvalidKey, errors.NewKV("Key", key))
```

Comment:
todo: Calling `WithStack` should not be required here.  All Defra errors already do this, the function is there to add stack stuff to non-defra errors :)

Reply:
just copied from another place. Will change